### PR TITLE
Attempt to fix endless exec wait

### PR DESCRIPTION
**Pull request by jasonbs10**
Attempt to fix endless exec wait. Increased flaky node probation from 1m->15m. Loosened mem_cap test requirements.

### CODE REVIEW

File: runner/execer/os/easy_test.go [hunk 1]
```diff
@@ -76,9 +76,9 @@ func TestOutput(t *testing.T) {
 }
 
 func TestMemUsage(t *testing.T) {
-	// Command to increase memory by 10MB every .1s until we hit 100MB after 1s.
+	// Command to increase memory by 1MB every .1s until we hit 50MB after 5s.
```
Comment (ryancouto) [inline runner/execer/os/easy_test.go:79]: What's the motivation behind these changes?

File: runner/execer/os/os_execer.go [hunk 2]
```diff
@@ -51,27 +51,49 @@ func (e *osExecer) Exec(command execer.Command) (result execer.Process, err erro
...
+	stdErrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return nil, err
+	}
+	stdOutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, err
+	}
+	var wg sync.WaitGroup
+	wg.Add(2)
```
Comment (ryancouto) [inline runner/execer/os/os_execer.go:80]: super duper nit: might be more clear to wg.Add(1) in each goroutine

**Review by ryancouto — APPROVED** [review]

File: runner/execer/os/easy_test.go (hunk 1 above)
Comment (jasonbs10) [reply runner/execer/os/easy_test.go:79]: These tests fail when my macbook has a high load.

File: runner/execer/os/os_execer.go (hunk 2 above)
Comment (jasonbs10) [reply runner/execer/os/os_execer.go:80]: I believe it needs to be done outside of the goroutines.
